taxcalc/records.py

```diff
@@ -25,31 +32,178 @@ class Records(object):
+                'c04500', '_amtstd', '_oldfei', 'c05200', '_cglong',
+                '_noncg', '_hasgain', '_dwks9', '_dwks5', '_dwks12',
+                '_dwks16', '_dwks17', '_dwks21', '_dwks25', '_dwks26',
+                '_dwks28', '_dwks31', 'c24505', 'c24510', 'c24520',
+                'c24530', 'c24540', 'c24534', 'c24597', 'c24598',
+                'c24610', 'c24615', 'c24550', 'c24570', '_addtax',
+                'c24560', '_taxspecial', 'c24580', 'c05100', 'c05700',
+                'c59430', 'c59450', 'c59460', '_line17', '_line19',
+                '_line22', '_line30', '_line31', '_line32', '_line36',
+                '_line33', '_line34', '_line35', 'c59485', 'c59490',
+                '_s1291', '_parents', 'c62720', 'c60260', 'c63100',
+                'c60200', 'c60240', 'c60220', 'c60130', 'c62730',
+                '_addamt', 'c62100', '_cmbtp', '_edical', '_amtsepadd',
+                '_agep', '_ages', 'c62600', 'c62700', '_alminc',
+                '_amtfei', 'c62780', 'c62900', 'c63000', 'c62740',
+                '_ngamty', 'c62745', 'y62745', '_tamt2', '_amt5pc',
+                '_amt15pc', '_amt25pc', 'c62747', 'c62755', 'c62770',
+                '_amt', 'c62800', 'c09600', 'c05800', '_ncu13',
+                '_seywage', 'c33465', 'c33470', 'c33475', 'c33480',
+                'c32840', '_tratio', 'c33200', 'c33400', 'c07180',
+                '_ieic', 'EICYB1', 'EICYB2', 'EICYB3', '_modagi',
+                '_val_ymax', '_preeitc', '_val_rtbase', '_val_rtless',
+                '_dy', 'c11070', '_nctcr', '_ctcagi', 'c87482',
+                'c87487', 'c87492', 'c87497', 'c87483', 'c87488',
+                'c87493', 'c87498', 'c87540', 'c87530', 'c87654',
+                'c87656', 'c87658', 'c87660', 'c87662', 'c87664',
+                'c87666', 'c10960', 'c87668', 'c87681', 'c87560',
+                'c87570', 'c87580', 'c87590', 'c87600', 'c87610',
+                'c87620', '_ctc1', '_ctc2', '_regcrd', '_exocrd',
+                '_ctctax', 'c07220', 'c82925', 'c82930', 'c82935',
+                'c82880', 'h82880', 'c82885', 'c82890', 'c82900',
+                'c82905', 'c82910', 'c82915',  'c82920', 'c82937',
+                'c82940', 'c11070', 'e59660', '_othadd', 'y07100',
+                'x07100', 'c08800', 'e08795', 'x07400', 'c59680',
+                '_othertax', 'e82915', 'e82940', 'SFOBYR', 'NIIT',
+                'c59720', '_comb', 'c07150', 'c10300', '_ospctax',
+                '_refund', 'c11600', 'e11450', 'e82040', 'e11500',
+                '_amed', '_xlin3', '_xlin6', 'share_corptax_burden',
+                'expanded_income', 'agg_self_employed', 'netcapgains',
+                'agg_capgains', 'total_compensation', 'dividends',
+                'agg_dividends', 'bonds', 'compensation', 'agg_bonds',
+                '_expanded_income']
+
+        if data is not None:
+            self.read(data)
+        else:
+            assert type(dims) is not None
```

Comment:
I'm not sure this is what you want here. Even if `dims` is `None` (which is the default value), the type of `dims` is a special type, `NoneType` which is not `None` so this assertion will pass. In fact, I can't think of any case where this assertion would fail (since even `None` has a type), so I don't think the assertion checks for any useful case. What is the desired check here?